Persistence/Persistence/Souces/FilePersistence.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  FilePersistence.swift
+//  Persistence
+//
+//  Created by 박승찬 on 11/20/24.
+//
+
+import DataSource
+import Foundation
+import OSLog
+
+public struct FilePersistence: FilePersistenceInterface {
+    private let fileManager = FileManager.default
+    private var documentDirectoryURL: URL? {
+        fileManager.urls(for: .documentDirectory, in: .userDomainMask).first
+    }
+    private let logger = Logger()
+
+    public func save(dataInfo: DataInformationDTO, data: Data) -> URL? {
+        guard let directoryURL = documentDirectoryURL?
+            .appendingPathComponent(
+                dataInfo
+                    .type
+                    .directoryName)
+        else { return nil }
+
+        createDirectory(at: directoryURL.path)
+
+        return write(to: directoryURL, with: data, fileName: dataInfo.identifier.uuidString)
+    }
+
+    public func load(path: URL) -> Data? {
+        var data: Data?
+
+        do {
+            data = try Data(contentsOf: path)
+        } catch {
+            logger.log("FilePersistence: 데이터 읽기 실패 \(path)")
+        }
+
+        return data
```

Comment:
data 선언을 do 전에 해주는 이유는 뭔가요 ??   
do 안에서 return 바아루 해주면 안되나요 ?

Reply:
크게 상관 없습니다!
그렇게 될 경우 return 문에서 nil을 반환해 줘야 하는데요,

어떤 것이 더 가독성이 좋을까요??
```suggestion
        do {
            return try Data(contentsOf: path)
        } catch {
            logger.log("FilePersistence: 데이터 읽기 실패 \(path)")
        }

        return nil
```